accepted/2021/package-scoring/PackageScoring.md

```diff
@@ -0,0 +1,259 @@
+
+There's many challenges to the quality of a package such as how many dependencies the package takes on, the amount of downloads the package has, and being maintained by a large company to name a few.
+
+NuGet has many different types of packages such as frameworks, dependencies, tools, templates, and many more. With each package, there is metadata & package contents that are used to help define the overall package.
+
+A high quality or "healthy" package is one that follows the following characteristics:
+
+- **It is actively maintained.** Either with recent commits or an annual update/notice that the package is up-to-date.
+- **It is documented.** It provides enough documentation to install, get started, and has public API documentation on it's members.
+- **It has a way to report bugs.** It provides a centralized location or repository in which issues are regularly triaged & resolved in future releases.
+- **It resolves security flaws quickly.** It fixes known vulnerabilities & releases an unaffected release quickly.
+- **It is not deprecated.** It is in an active state meeting all the criteria above.
+
+This is a small list of the many different characteristics that may make a package of high quality or "healthy".
+
+Having packages of high quality that depend on other packages of high quality will make the overall software supply chain of higher quality and thus more secure & reliable for those using .NET tooling.
+
+This proposal introduces the concept known as Package Scoring or NET Score for short.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+Developers are frustrated with the packages on NuGet meeting their needs. These needs are typically categorized in multiple categories:
+
+1. **Popularity**- How popular a package is & recognized in the ecosystem.
+2. **Quality** - The completeness of a package following best practices & providing documentation.
+3. **Maintenance** - The state of maintenance & free of security vulnerabilities.
+4. **Community** - The sustainability of an active community for the package.
+
+There are a number of problems in each of these categories.
+
+- **Popularity**
+  - Number of downloads is used as a proxy to the overall quality of the package. 
+    - If other people download it, that means it must be good!
+  - Additional metrics should be added such as how many packages depend on the package. 
+    - A network of what depends on the package may make it more practical to the ecosystem.
+  - Popular packages may take on many dependencies that make a popular package “unhealthy”.
+    - The more dependencies a popular package takes on & isn’t maintained, the higher likelihood more people become vulnerable from downloading it.
+- **Quality**
```

Comment:
I personally believe nuget packages (at least not preview ones) should only include assemblies should prefer performance over debuggability. More specifically be compiled with optimization. 

If we agree on that, it would make sense to score packages accordingly. ([shameless plug on the topic](https://garcia.in/posts/2021/pitfalls-of-unoptimized-nuget-packages/))